### PR TITLE
DPDK Backend: Update DPDK Readme.md

### PR DESCRIPTION
**Pull request by usha1830**

### CODE REVIEW

File: backends/dpdk/README.md [hunk 1]
```diff
@@ -20,17 +22,48 @@ generates the 'spec' file to configure the DPDK pipeline.
...
+- Direct Meter
+- Direct Counter
+- Basic Checksum
+- Packet Cloning/Recirculation/Resubmission
+
+### DPDK target limitations
+- Currently, programs written for DPDK target should limit the functionality in Ingress blocks.  When egress block support gets added to the DPDK target, compiler should generate separate spec file for ingress and egress.
+- DPDK architecture assumes the following signatures for programmable block of PSA. P4C-DPDK converts the input program to this form.
+
+```bash
```
Comment (rst0git) [inline backends/dpdk/README.md:51]: GitHub supports syntax highlighting for P4, you can use it with ```P4

File: backends/dpdk/README.md [hunk 2]
```diff
@@ -20,17 +22,48 @@ generates the 'spec' file to configure the DPDK pipeline.
...
+
+parser EgressParser (packet_in buffer, out H parsed_hdr, inout M user_meta);
+
+control Egress (inout H hdr, inout M user_meta);
+
+control EgressDeparser (packet_out buffer, inout H hdr, in M meta);
+```
 
+- Size of all structure fields to be multiple of 8 bits and should not be greater than 64-bit. (This limitation is planned to be removed by grouping fields into fields which are multiple of 8-bits)
+- Combination of header and metadata as table key is not allowed. (This limitation is removed by P4C by copying the differing match key fields to metadata)
```
Comment (jafingerhut) [inline backends/dpdk/README.md:66]: "is removed" meaning "is planned to be removed", similar to the first restrction?

Both of these restrictions seem pretty limiting for what typical P4 targets support, so great to hear there is a plan to remove these restrictions.

Comment (hanw) [reply backends/dpdk/README.md:66]: Combining header and metadata as table key is already implemented by Usha.

**Review by hanw — APPROVED** [review]

File: backends/dpdk/README.md (hunk 2 above)
Comment (jafingerhut) [reply backends/dpdk/README.md:66]: Already implemented before this change to the README and merged in?  Or planned to be merged by a future PR?

If it is already implemented before this PR, then why list it as a limitation in the README?

Comment (hanw) [reply backends/dpdk/README.md:66]: DPDK pipeline still has this limitation, but compiler was able to hide the limitation from p4 programmer by copying header field to metadata prior to table lookup.

Comment (jafingerhut) [reply backends/dpdk/README.md:66]: Understood.  Makes sense now.

**Review by jafingerhut — APPROVED** [review]
LGTM

**Review by rst0git — APPROVED** [review]

**Pull request merged**